### DISCUSSION TITLE
Automatically depend on NEXT_PUBLIC env vars

### DISCUSSION THREAD

**Discussion by ericmatthys** [vercel/turborepo#981] [Ideas]
Given the out of the box support with Vercel and somewhat by extension Next.js, I think it'd be worth considering assuming all `NEXT_PUBLIC` env vars affect the hash used for caching. We have quite a few and now our `turbo.json` is another place we need to track them all. Given that `NEXT_PUBLIC` env vars are used for vars that can baked into in the build, I think it'd be a safe assumption that any env var that begins with `NEXT_PUBLIC` should affect the hash. I believe this would lead to fewer issues from the Vercel community giving Turborepo a try.

**Comment by ericmatthys** [comment]
Alternatively, perhaps env vars in `dependsOn` could support a wildcard character, like `NEXT_PUBLIC_*`

**Comment by jaredpalmer** [comment]
Love it.

**Comment by anthonyshew** [comment]
Framework Inference is now properly documented here: https://turbo.build/repo/docs/crafting-your-repository/using-environment-variables#framework-inference
